### PR TITLE
Use containerd client for container execsync

### PR DESCRIPTION
**Pull request by Random-Liu**
I'll be using new containerd client when implementing containerd streaming, so update container execsync first.

/cc @abhinandanpb 

Note that I'm not using `process.Wait`. Will use it after https://github.com/containerd/containerd/pull/1287 and https://github.com/containerd/containerd/pull/1279 are merged.

### CODE REVIEW

File: pkg/server/container_execsync.go [hunk 1]
```diff
@@ -48,92 +43,68 @@ func (c *criContainerdService) ExecSync(ctx context.Context, r *runtime.ExecSync
...
-		ExecID:      execID,
-	}); err != nil && !isContainerdGRPCNotFoundError(err) {
-		return nil, fmt.Errorf("failed to delete exec %q in container %q: %v", execID, id, err)
-	}
 	// TODO(random-liu): [P1] Deal with timeout, kill and wait again on timeout.
 
-	// TODO(random-liu): Make sure stdout/stderr are drained.
+	// Wait for the io to be drained.
+	task.IO().Wait()
+
```
Comment (mikebrow) [inline pkg/server/container_execsync.go]: task.IO().Close() ?

Comment (Random-Liu) [inline pkg/server/container_execsync.go]: That's included in `task.Delete`.

File: pkg/server/container_execsync.go [hunk 2]
```diff
@@ -48,92 +43,68 @@ func (c *criContainerdService) ExecSync(ctx context.Context, r *runtime.ExecSync
...
+	process, err := task.Exec(ctx, execID, pspec, io)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create exec streaming directory: %v", err)
+		return nil, fmt.Errorf("failed to create exec %q: %v", execID, err)
 	}
 	defer func() {
-		if err = c.os.RemoveAll(execDir); err != nil {
-			glog.Errorf("Failed to remove exec streaming directory %q: %v", execDir, err)
+		if _, err := process.Delete(ctx); err != nil {
+			glog.Errorf("Failed to delete exec process %q for container %q: %v", execID, id, err)
```
Comment (mikebrow) [inline pkg/server/container_execsync.go]: for sanity can we check the exit code from this Delete call against the exit code from c.waitContainerExec? Or am I missing something?

File: pkg/server/container_execsync.go (hunk 1 above)
Comment (mikebrow) [inline pkg/server/container_execsync.go]: i see then probably just need to close if there's an error on task.exec above..

File: pkg/server/container_execsync.go [hunk 3]
```diff
@@ -48,92 +43,68 @@ func (c *criContainerdService) ExecSync(ctx context.Context, r *runtime.ExecSync
...
-		},
-		ExecID: execID,
-	})
-	if err != nil {
-		return nil, fmt.Errorf("failed to exec in container %q: %v", id, err)
+	if err := process.Start(ctx); err != nil {
+		return nil, fmt.Errorf("failed to start exec %q: %v", execID, err)
 	}
+
 	exitCode, err := c.waitContainerExec(eventstream, id, execID)
```
Comment (mikebrow) [inline pkg/server/container_execsync.go]: should this be process.Wait() now?

File: pkg/server/container_execsync.go (hunk 1 above)
Comment (Random-Liu) [inline pkg/server/container_execsync.go]: `Exec` has handled cleanup on error inside.

File: pkg/server/container_execsync.go (hunk 2 above)
Comment (Random-Liu) [inline pkg/server/container_execsync.go]: Seems not necessary. :)

File: pkg/server/container_execsync.go (hunk 3 above)
Comment (Random-Liu) [inline pkg/server/container_execsync.go]: Let's use `Wait` after we bump up containerd version including the fixes for `Wait`.

**Comment by Random-Liu** [comment]
@mikebrow The cleanup issue will be addressed in https://github.com/containerd/containerd/pull/1290.

File: pkg/server/container_execsync.go (hunk 3 above)
Comment (mikebrow) [inline pkg/server/container_execsync.go]: kk a // TODO then.

Comment (Random-Liu) [inline pkg/server/container_execsync.go]: There is one above.

File: pkg/server/container_execsync.go (hunk 2 above)
Comment (mikebrow) [inline pkg/server/container_execsync.go]: still a sanity check won't hurt... you can add this should not happen if you like to the error message that is un-necessary :-)

Comment (mikebrow) [inline pkg/server/container_execsync.go]: just makes me uneasy getting the exit code earlier than the Delete()

Comment (Random-Liu) [inline pkg/server/container_execsync.go]: Check ctr/run.go, or ctr/exec. They are doing similar thing.

Comment (Random-Liu) [inline pkg/server/container_execsync.go]: In the future , this will be Wait. Wait is guaranteed to return exit status if there is no error.

Comment (mikebrow) [inline pkg/server/container_execsync.go]: right and the delete invokes a request that also retrieves and exit status..

Sorry just my intuition here that there may come a day when one of the two ways to get the exit status will have an error or a different value and if so we'd want to investigate cause.

**Comment by Random-Liu** [comment]
@mikebrow Sorry for the delay. Went to take a nap just now. :p 
Apply LGTM based on https://github.com/kubernetes-incubator/cri-containerd/pull/111#pullrequestreview-54470601.

**Pull request merged**